docs/samples/istio-dex/README.md

```diff
@@ -5,102 +5,214 @@ This example shows how to create a KFServing InferenceService as well as sending
+
+_auth_session = get_istio_auth_session(
+    url=KUBEFLOW_ENDPOINT, username=KUBEFLOW_USERNAME, password=KUBEFLOW_PASSWORD
+)
+
+cookies = {"authservice_session": _auth_session['authservice_session']}
+jar = requests.cookies.cookiejar_from_dict(cookies)
+
+res = requests.post(
+    url=PREDICT_ENDPOINT,
+    headers={"Host": SERVICE_HOSTNAME, "Content-Type": "application/json"},
+    cookies=jar,
+    json=iris_input,
+    timeout=200,
+)
+print("Status Code: ", res.status_code)
+print("Response: ", res.json())
 ```
+> **NOTE:**
+> Change the Variables as necessary. 
+> 
+> You can get the cluster IP by executing the following command.
+> ```bash
+> kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.clusterIP}'
+> ```
+> 
+> You can get the `SERVICE_HOSTNAME` by executing the following command.
+> ```bash
+> kubectl get -n kubeflow-user-example-com inferenceservice ${MODEL_NAME} -o jsonpath='{.status.url}' | cut -d "/" -f 3
+> ```
 
-Expected Output
+### Expected Output
 
 ```bash
-*   Trying 10.152.183.241...
-* TCP_NODELAY set
-* Connected to 10.152.183.241 (10.152.183.241) port 80 (#0)
-> POST /v1/models/sklearn-iris:predict HTTP/1.1
-> Host: sklearn-iris.admin.example.com
-> User-Agent: curl/7.58.0
-> Accept: */*
-> Cookie: authservice_session=MTU4OTI5NDAzMHxOd3dBTkVveldFUlRWa3hJUVVKV1NrZE1WVWhCVmxSS05GRTFSMGhaVmtWR1JrUlhSRXRRUmtnMVRrTkpUekpOTTBOSFNGcElXRkU9fLgsofp8amFkZv4N4gnFUGjCePgaZPAU20ylfr8J-63T
-> Content-Length: 76
-> Content-Type: application/x-www-form-urlencoded
-> 
-* upload completely sent off: 76 out of 76 bytes
-< HTTP/1.1 200 OK
-< content-length: 23
-< content-type: text/html; charset=UTF-8
-< date: Tue, 12 May 2020 14:38:50 GMT
-< server: istio-envoy
-< x-envoy-upstream-service-time: 7307
-< 
-* Connection #0 to host 10.152.183.241 left intact
-{"predictions": [1, 1]}
+Status Code: 200
+Response: {"predictions": [1, 1]}
 ```
+### FAQ
+
+1. Why I am getting 404 not found error ?
+
+   Check the url and service hostname for typos. If you are sending request from outside the cluster and service 
+   hostname ends with `svc.cluster.local` domain, you need to [configure the domain/DNS](https://knative.dev/docs/install/yaml-install/serving/install-serving-with-yaml/#configure-dns)
+   in knative serving.
+
+2. Why I am getting 403 Forbidden RBAC access denied error ?
+   
+   If you are using istio sidecar injection then, disable it by adding the annotation `sidecar.istio.io/inject: false` to the InferenceService
+   or create an [Istio AuthorizationPolicy](https://istio.io/latest/docs/reference/config/security/authorization-policy/) to grant access to the pods.
+   Here is an example AuthorizationPolicy which grants access to the predictor pods.
+   ```yaml
+   apiVersion: security.istio.io/v1
+   kind: AuthorizationPolicy
+   metadata:
+     name: allow-predictor
+     namespace: istio-system
+   spec:
+     selector:
+       matchLabels:
+         component: predictor
```

Comment:
Should this be istio ingress gateway instead ?

Reply:
I think this policy just allows everything which should not be suggested to user